extensions/ui/org.eclipse.smarthome.ui.basic/src/main/java/org/eclipse/smarthome/ui/basic/internal/render/AbstractWidgetRenderer.java

```diff
@@ -66,10 +75,28 @@ public ItemUIRegistry getItemUIRegistry() {
         return itemUIRegistry;
     }
 
+    public void setLocaleProvider(LocaleProvider localeProvider) {
+        this.locale = localeProvider.getLocale();
```

Comment:
I'd keep the localeProvider and call `localeProvider.getLocale()` whenever you need it. Otherwise you won't notice if the user changes the locale settings.

Reply:
Makes ense. I realize we have the same mistake elsewhere, sometimes introduced by myself !